app/models/customer.rb

```diff
@@ -3,4 +3,15 @@ class Customer < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  validates_with CustomerEmailValidator
+  belongs_to :company
+  before_validation :register_company, on: :create
+
+  private
+
+  def register_company
+    domain = email.scan(/.+@(.+\..+)/).first.first
```

Comment:
Acho que podia trocar esse `first.first` por `flatten.first` só pra ficar mais claro o retorno de que é um array dentro de um array e você quer simplificar, o que acha?

```suggestion
    domain = email.scan(/.+@(.+\..+)/).flatten.first
```